### PR TITLE
Added missing annotations for index checker in ResultSetMetaData.java

### PR DESCRIPTION
**Pull request by PRITI1999**

### CODE REVIEW

File: src/java.sql/share/classes/java/sql/ResultSetMetaData.java [hunk 1]
```diff
@@ -43,7 +43,8 @@
  *
  * @since 1.1
  */
-
```
Comment (mernst) [inline src/java.sql/share/classes/java/sql/ResultSetMetaData.java]: Please don't make whitespace changes.

File: src/java.sql/share/classes/java/sql/ResultSetMetaData.java [hunk 2]
```diff
@@ -43,7 +43,8 @@
...
+import org.checkerframework.checker.index.qual.NonNegative;
```
Comment (mernst) [inline src/java.sql/share/classes/java/sql/ResultSetMetaData.java]: Import statements should go with the `package` statement, not between the class Javadoc and the class definition.

File: src/java.sql/share/classes/java/sql/ResultSetMetaData.java [hunk 3]
```diff
@@ -135,7 +136,7 @@
      *          of the designated column
      * @exception SQLException if a database access error occurs
      */
-    int getColumnDisplaySize(int column) throws SQLException;
+    @NonNegative int getColumnDisplaySize(@Positive int column) throws SQLException;
```
Comment (mernst) [inline src/java.sql/share/classes/java/sql/ResultSetMetaData.java:142]: I found multiple different implementations of this interface that sometimes return 0.  Are you sure that all those implementations are incorrect?  What are you basing this `@NonNegative` annotation on?

File: src/java.sql/share/classes/java/sql/ResultSetMetaData.java [hunk 4]
```diff
@@ -180,7 +181,7 @@
      * @return precision
      * @exception SQLException if a database access error occurs
      */
-    int getPrecision(int column) throws SQLException;
+    @NonNegative int getPrecision(@Positive int column) throws SQLException;
```
Comment (mernst) [inline src/java.sql/share/classes/java/sql/ResultSetMetaData.java:187]: The Javadoc explicitly says that the return value may be 0, on line 177 above.  Why did you annotate the return type as `@NonNegative`?

**Review by mernst — CHANGES_REQUESTED** [review]

File: src/java.sql/share/classes/java/sql/ResultSetMetaData.java (hunk 3 above)
Comment (PRITI1999) [reply src/java.sql/share/classes/java/sql/ResultSetMetaData.java:142]: `@NonNegative`'s documentation says "The annotated expression evaluates to an integer greater than or equal to 0. " which means that the return value includes 0. Did I get the usage of the annotation wrong?

Comment (mernst) [reply src/java.sql/share/classes/java/sql/ResultSetMetaData.java:142]: Sorry, I switched around the two annotations on this line.  I apologize for my mistake.

**Comment by mernst** [comment]
@PRITI1999 The pull request is failing because you have not pulled master into the master branch of your fork of `checker-framework`.  Could you do so?  Thanks.

**Comment by PRITI1999** [comment]
@mernst Updated my master of `checker-framework`.

**Pull request merged**